merkle-tree/concurrent/tests/tests.rs

```diff
@@ -1212,3 +1215,125 @@ fn test_changelog_event_v1() {
+        let mut onchain_merkle_tree =
+            ConcurrentMerkleTree::<Poseidon, 26>::new(26, 10, 10, 10).unwrap();
+        onchain_merkle_tree.init().unwrap();
+        assert_eq!(
+            onchain_merkle_tree.root().unwrap(),
+            crank_merkle_tree.root()
+        );
+        // nullified
+        let leaf_payer = [
+            13, 210, 199, 72, 48, 165, 14, 149, 233, 152, 84, 206, 101, 234, 164, 81, 145, 231,
+            254, 25, 53, 13, 61, 113, 165, 121, 130, 242, 113, 28, 169, 160,
+        ];
+        onchain_merkle_tree.append(&leaf_payer).unwrap();
+        crank_merkle_tree.append(&leaf_payer);
+
+        // the first will be nullified in second transfer
+        let leaf_after_transfer = [
+            [
+                8, 20, 136, 212, 52, 210, 10, 51, 118, 173, 8, 211, 29, 52, 104, 190, 86, 102, 232,
+                175, 95, 23, 26, 1, 61, 35, 193, 87, 25, 103, 2, 168,
+            ],
+            [
+                47, 94, 164, 111, 248, 69, 228, 25, 157, 196, 219, 250, 206, 26, 127, 51, 115, 89,
+                43, 126, 33, 67, 157, 57, 2, 250, 105, 73, 182, 13, 166, 221,
+            ],
+        ];
+        onchain_merkle_tree
+            .append(&leaf_after_transfer.first().unwrap())
+            .unwrap();
+        onchain_merkle_tree
+            .append(&leaf_after_transfer.last().unwrap())
+            .unwrap();
+
+        crank_merkle_tree
+            .append(&leaf_after_transfer.first().unwrap())
+            .unwrap();
+        crank_merkle_tree
+            .append(&leaf_after_transfer.last().unwrap())
+            .unwrap();
+
+        let leaf_after_transfer_2 = [
+            [
+                3, 112, 217, 207, 229, 87, 74, 103, 34, 107, 18, 158, 161, 200, 146, 23, 66, 161,
+                252, 37, 160, 174, 60, 72, 196, 228, 44, 242, 32, 49, 112, 39,
+            ],
+            [
+                38, 231, 184, 31, 11, 199, 22, 51, 105, 158, 5, 174, 9, 241, 129, 75, 74, 173, 147,
+                178, 69, 9, 31, 178, 219, 154, 139, 158, 27, 109, 208, 88,
+            ],
+        ];
+        onchain_merkle_tree
+            .append(&leaf_after_transfer_2.first().unwrap())
+            .unwrap();
+        onchain_merkle_tree
+            .append(&leaf_after_transfer_2.last().unwrap())
+            .unwrap();
+        crank_merkle_tree
+            .append(&leaf_after_transfer_2.first().unwrap())
+            .unwrap();
+        crank_merkle_tree
+            .append(&leaf_after_transfer_2.last().unwrap())
+            .unwrap();
+        // we should have a queue with 2 elements leaf payer leaf_after_transfer[0]
+        let nullifier_1 = BigUint::from_be_bytes(leaf_payer.as_slice());
+        let nullifier_2 = BigUint::from_be_bytes(leaf_after_transfer.first().unwrap());
+        let mut queue = HashSet::<u16>::new(6857, 4800, 2400).unwrap();
+        queue.insert(&nullifier_1, 1).unwrap();
+        queue.insert(&nullifier_2, 1).unwrap();
```

Comment:
can you remove the ugly hardcoded leaves?
(the test should work the same way without.)